python/src/aiconfig/eval/metrics.py

```diff
@@ -29,22 +29,65 @@ async def __call__(self, datum: common.T_Evaluable) -> common.T_MetricValue:
         return await self.evaluation_fn(datum)
 
 
-def _check_substring(
-    output_datum: str,
-    substring: str,
-    #
-    case_sensitive: bool,
-) -> bool:
-    if case_sensitive:
-        return substring in output_datum
-    else:
-        return substring.lower() in output_datum.lower()
+PS = ParamSpec("PS")
 
 
-async def _calculate_brevity(datum: str) -> int:
-    if len(datum) == 0:
-        raise ValueError("Brevity is meaningless for empty string.")
-    return len(datum)
+@core_utils.parametrized
+def metric(
+    parametrized_evaluation_fn: Callable[Concatenate[common.T_Evaluable, PS], common.T_MetricValue],
+    name: str | None = None,
+    description: str | None = None,
+    best_value: common.T_MetricValue | None = None,
+    worst_value: common.T_MetricValue | None = None,
+) -> Callable[PS, Metric[common.T_Evaluable, common.T_MetricValue]]:
+    name_ = name or parametrized_evaluation_fn.__name__
+    description_ = description or name_
```

Comment:
nit, would probably be good to differentiate the description text from name. Maybe default to f"{name} evaluation metric"?